apps/heft/src/plugins/JestPlugin/JestTypeScriptDataFile.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { JsonFile } from '@rushstack/node-core-library';
+import { ITypeScriptConfiguration } from '../../stages/BuildStage';
+
+/**
+ * Schema for jest-typescript-data.json
+ */
+export interface IJestTypeScriptDataFileJson {
+  /**
+   * The "emitFolderPathForJest" from .heft/typescript.json
+   */
+  emitFolderPathForJest: string;
+}
+
+/**
+ * Manages loading/saving the "jest-typescript-data.json" data file.  This file communicates
+ * configuration information from Heft to jest-build-transform.js.  The jest-build-transform.js script gets
+ * loaded dynamically by the Jest engine, so it does not have access to the normal HeftConfiguration objects.
+ */
+export class JestTypeScriptDataFile {
+  /**
+   * Called by TypeScriptPlugin to write the file.
+   */
+  public static saveForProject(
+    projectFolder: string,
+    typeScriptConfiguration: ITypeScriptConfiguration
+  ): void {
+    const jsonFilePath: string = JestTypeScriptDataFile.getConfigFilePath(projectFolder);
+
+    const json: IJestTypeScriptDataFileJson = {
+      emitFolderPathForJest: typeScriptConfiguration.emitFolderPathForJest || 'lib'
+    };
+    JsonFile.save(json, jsonFilePath);
+  }
+
+  /**
+   * Called by jest-build-transform.js to read the file.
+   */
+  public static loadForProject(projectFolder: string): IJestTypeScriptDataFileJson {
+    const jsonFilePath: string = JestTypeScriptDataFile.getConfigFilePath(projectFolder);
+    return JsonFile.load(jsonFilePath);
```

Comment:
Do you want to include a schema file for this? Or do you think it probably doesn't matter because this is never expected to be human-edited?

Reply:
It's an "internal" file that is only used by Heft itself, so I think we can trust the file contents to be correct.  

Also, if we can improve the config file APIs so that the transform can load TypeScript.json directly, then it's likely we will eliminate this file entirely.

Reply:
I added notice at the top of the file to make it clear that it is "internal."  We've had problems in the past with people loading and relying on data structures that were not intended to be public.